File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        int max = -1;
        Arrays.sort(nums);
        
        for(int i = 0, j = nums.length-1; i < nums.length/2; i++,j--){
            if(nums[i] + nums[j] > max)
                max = nums[i] + nums[j];
        }
        return max;
    }
}

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.swift
class Solution {
    func minPairSum(_ nums: [Int]) -> Int {
        var tmp = nums.sorted(by: <)
        var max = -1
        var i = 0
        var j = tmp.count-1
        
        while i < tmp.count/2 {
            if tmp[i] + tmp[j] > max { max = tmp[i] + tmp[j] }
            i += 1
            j -= 1
        }
        
        return max
    }
}